tensorflow/python/ops/nn_test.py

```diff
@@ -179,23 +179,25 @@ class L2LossTest(test_lib.TestCase):
 
   def testL2Loss(self):
     for dtype in [dtypes.float32, dtypes.float64]:
-      with self.test_session():
-        x = constant_op.constant(
-            [1.0, 0.0, 3.0, 2.0], shape=[2, 2], name="x", dtype=dtype)
-        l2loss = nn_ops.l2_loss(x)
-        value = l2loss.eval()
-      self.assertAllClose(7.0, value)
+      for test_gpu in [ False, True ]:
```

Comment:
In other tests I've seen a `(type, use_gpu)` pair used, either with a `TestConfig` helper function (like [here](https://github.com/lukeiwanski/tensorflow/blob/f48673b5054b474fa1e51823edd075088cd16d5f/tensorflow/python/kernel_tests/pooling_ops_3d_test.py#L32) ) or just using `itertools.product` (like [here](https://github.com/lukeiwanski/tensorflow/blob/f48673b5054b474fa1e51823edd075088cd16d5f/tensorflow/python/training/training_ops_test.py#L125) ). This then means you don't have to re-indent everything which minimises changes and helps to avoid errors like the one below.